pkg/synopsysctl/cmd_create.go

```diff
@@ -358,103 +292,58 @@ var createBlackDuckCmd = &cobra.Command{
-			cmd.Flag("pvc-file-path").Changed = false // we already did the special logic above to set the PVCs
-			blackDuck, err := updateBlackDuckSpecWithFlags(cmd, blackDuckName, blackDuckNamespace)
-			if err != nil {
-				return err
-			}
+		fmt.Printf("Helm Map\n%+v\n", helmValuesMap)
 
-			// add versions
-			if len(blackDuck.Spec.Version) == 0 {
-				// versions := apps.NewApp(&protoform.Config{}, restconfig).Blackduck().Versions()
-				// sort.Sort(sort.Reverse(sort.StringSlice(versions)))
-				// TODO: fix the sort logic for Black Duck version
-				blackDuck.Spec.Version = "2020.2.1"
-			}
-			versionSupportsSecurityContexts, err := util.IsVersionGreaterThanOrEqualTo(blackDuck.Spec.Version, 2019, time.December, 0)
-			if err != nil {
-				return fmt.Errorf("failed to check Black Duck version: %s", err)
-			}
-			if !versionSupportsSecurityContexts && cmd.Flags().Changed("security-context-file-path") {
-				log.Warnf("security contexts from --security-context-file-path are ignored for versions before 2019.12.0, you're using version %s", blackDuck.Spec.Version)
+		// Update the Helm Chart Location
+		chartLocationFlag := cmd.Flag("chart-location-path")
+		if chartLocationFlag.Changed {
+			blackduckChartRepository = chartLocationFlag.Value.String()
+		} else {
+			versionFlag := cmd.Flag("version")
+			if versionFlag.Changed {
+				blackduckChartRepository = fmt.Sprintf("https://artifactory.internal.synopsys.com/artifactory/bds-hub-helm-snapshot-local/blackduck/blackduck-%s.tgz", versionFlag.Value.String())
 			}
-
-			return PrintResource(*blackDuck, mockFormat, false)
 		}
 
-		blackDuck, err := updateBlackDuckSpecWithFlags(cmd, blackDuckName, blackDuckNamespace)
+		secrets, err := blackduck.GetCertsFromFlagsAndSetHelmValue(args[0], namespace, cmd.Flags(), helmValuesMap)
 		if err != nil {
 			return err
 		}
-
-		log.Infof("creating Black Duck '%s' in namespace '%s'...", blackDuckName, blackDuckNamespace)
-		if len(blackDuck.Spec.Version) == 0 {
-			// versions := apps.NewApp(&protoform.Config{}, restconfig).Blackduck().Versions()
-			// sort.Sort(sort.Reverse(sort.StringSlice(versions)))
-			// TODO: fix the sort logic for Black Duck version
-			blackDuck.Spec.Version = "2020.2.1"
-		}
-		versionSupportsSecurityContexts, err := util.IsVersionGreaterThanOrEqualTo(blackDuck.Spec.Version, 2019, time.December, 0)
-		if err != nil {
-			return fmt.Errorf("failed to check Black Duck version: %s", err)
-		}
-		if !versionSupportsSecurityContexts && cmd.Flags().Changed("security-context-file-path") {
-			return fmt.Errorf("security contexts from --security-context-file-path cannot be set for versions before 2019.12.0, you're using version %s", blackDuck.Spec.Version)
-		}
-		if util.IsOpenshift(kubeClient) && cmd.Flags().Changed("security-context-file-path") {
-			return fmt.Errorf("cannot set security contexts with --security-context-file-path in an Openshift environment")
+		for _, v := range secrets {
```

Comment:
For Polaris-Reporting we used `KubectlApplyRuntimeObjects` so that it doesn't care about the type of resource. I think this is okay tho :) and it helps us move away from being dependent on the kubectl binary. I'm going to create a ticket to make a function that creates any resource with the KubeClient